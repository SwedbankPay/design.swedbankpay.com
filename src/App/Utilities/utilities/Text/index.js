import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const TextAlignment = () => (
    <>
        <h2 id="text-alignment">Text alignment</h2>
        <p>Easily realign text to components with text alignment classes.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p className="text-justify">
                Ambitioni dedisse scripsisse iudicaretur. Cras mattis iudicium purus sit amet fermentum. Donec sed odio operae, eu vulputate felis rhoncus. Praeterea iter est quasdam res quas ex communi. At nos hinc posthac, sitientis piros Afros. Petierunt uti sibi concilium totius Galliae in diem certam indicere. Cras mattis iudicium purus sit amet fermentum.
            </p>
        </ComponentPreview>

        <p>For left, right, and center alignment, responsive classes are available that use the same viewport width breakpoints as the grid system.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p className="text-left">Left aligned text on all viewport sizes.</p>
            <p className="text-center">Center aligned text on all viewport sizes.</p>
            <p className="text-right">Right aligned text on all viewport sizes.</p>
            {"\n\n"}
            <p className="text-sm-left">Left aligned text on viewports sized SM (small) or wider.</p>
            <p className="text-md-left">Left aligned text on viewports sized MD (medium) or wider.</p>
            <p className="text-lg-left">Left aligned text on viewports sized LG (large) or wider.</p>
            <p className="text-xl-left">Left aligned text on viewports sized XL (extra-large) or wider.</p>
            <p className="text-xxl-left">Left aligned text on viewports sized XXL (extra-extra-large) or wider.</p>
        </ComponentPreview>

    </>
);

const TextWrappingAndOverflow = () => (
    <>
        <h2 id="text-wrapping-and-overflow">Text wrapping and overflow</h2>
        <p>Prevent text from wrapping with a <CodeTags type="secondary" code=".text-nowrap" /> class.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="text-nowrap sc-highlight" style={{ width: "8rem" }}>{"\n"}
                This text should overflow the parent.{"\n"}
            </div>
        </ComponentPreview>

        <p>For longer content, you can add a <CodeTags type="secondary" code=".text-truncate" /> class to truncate the text with an ellipsis. <b>Requires</b> display: <CodeTags type="secondary" code="inline-block" /> <b>or</b> <CodeTags type="secondary" code="display: block" />.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="row">
                <div className="col-3 text-truncate">
                    Cras egestas dolor nec nunc scelerisque tincidunt.
                </div>
            </div>{"\n\n"}

            <span className="d-inline-block text-truncate" style={{ maxWidth: "150px" }}>{"\n"}
                Cras egestas dolor nec nunc scelerisque tincidunt.{"\n"}
            </span>
        </ComponentPreview>
    </>
);

const TextTransform = () => (
    <>
        <h2 id="text-transform">Text transform</h2>
        <p>Transform text in components with text capitalization classes.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p className="text-lowercase">Lowercased text.</p>
            <p className="text-uppercase">Uppercased text.</p>
            <p className="text-capitalize">CapiTaliZed text.</p>
        </ComponentPreview>
        <p>Note how <CodeTags type="secondary" code=".text-capitalize" /> only changes the first letter of each word, leaving the case of any other letters unaffected.</p>
    </>
);

const FontWeightAndItalics = () => (
    <>
        <h2 id="font-weight-and-italics">Font weight and italics</h2>
        <p>Quickly change the weight (boldness) of text or italicize text.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p className="font-weight-bold">Bold text.</p>
            <p className="font-weight-normal">Normal weight text.</p>
            <p className="font-weight-light">Light weight text.</p>
            <p className="font-italic">Italic text.</p>
        </ComponentPreview>
    </>
);

const Text = () => (
    <DocContainer docToc>
        <p className="lead">Documentation and examples for common text utilities to control alignment, wrapping, weight, and more.</p>
        <TextAlignment />
        <TextWrappingAndOverflow />
        <TextTransform />
        <FontWeightAndItalics />
    </DocContainer>
);

export default Text;

/* For testing */
export { TextAlignment, TextWrappingAndOverflow, TextTransform, FontWeightAndItalics };
