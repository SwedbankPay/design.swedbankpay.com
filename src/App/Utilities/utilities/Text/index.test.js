import React from "react";
import { shallow } from "enzyme";

import Text, { TextAlignment, TextWrappingAndOverflow, TextTransform, FontWeightAndItalics } from "./index";

describe("Utilities: Text", () => {
    it("is defined", () => {
        expect(Text).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Text />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("TextAlignment", () => {
        it("is defined", () => {
            expect(TextAlignment).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<TextAlignment />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("TextWrappingAndOverflow", () => {
        it("is defined", () => {
            expect(TextWrappingAndOverflow).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<TextWrappingAndOverflow />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("TextTransform", () => {
        it("is defined", () => {
            expect(TextTransform).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<TextTransform />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("FontWeightAndItalics", () => {
        it("is defined", () => {
            expect(FontWeightAndItalics).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<FontWeightAndItalics />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
