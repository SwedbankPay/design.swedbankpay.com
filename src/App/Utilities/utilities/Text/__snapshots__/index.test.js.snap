// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: Text FontWeightAndItalics renders 1`] = `
<Fragment>
  <h2
    id="font-weight-and-italics"
  >
    Font weight and italics
  </h2>
  <p>
    Quickly change the weight (boldness) of text or italicize text.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p
      className="font-weight-bold"
    >
      Bold text.
    </p>
    <p
      className="font-weight-normal"
    >
      Normal weight text.
    </p>
    <p
      className="font-weight-light"
    >
      Light weight text.
    </p>
    <p
      className="font-italic"
    >
      Italic text.
    </p>
  </ComponentPreview>
</Fragment>
`;

exports[`Utilities: Text TextAlignment renders 1`] = `
<Fragment>
  <h2
    id="text-alignment"
  >
    Text alignment
  </h2>
  <p>
    Easily realign text to components with text alignment classes.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p
      className="text-justify"
    >
      Ambitioni dedisse scripsisse iudicaretur. Cras mattis iudicium purus sit amet fermentum. Donec sed odio operae, eu vulputate felis rhoncus. Praeterea iter est quasdam res quas ex communi. At nos hinc posthac, sitientis piros Afros. Petierunt uti sibi concilium totius Galliae in diem certam indicere. Cras mattis iudicium purus sit amet fermentum.
    </p>
  </ComponentPreview>
  <p>
    For left, right, and center alignment, responsive classes are available that use the same viewport width breakpoints as the grid system.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p
      className="text-left"
    >
      Left aligned text on all viewport sizes.
    </p>
    <p
      className="text-center"
    >
      Center aligned text on all viewport sizes.
    </p>
    <p
      className="text-right"
    >
      Right aligned text on all viewport sizes.
    </p>
    


    <p
      className="text-sm-left"
    >
      Left aligned text on viewports sized SM (small) or wider.
    </p>
    <p
      className="text-md-left"
    >
      Left aligned text on viewports sized MD (medium) or wider.
    </p>
    <p
      className="text-lg-left"
    >
      Left aligned text on viewports sized LG (large) or wider.
    </p>
    <p
      className="text-xl-left"
    >
      Left aligned text on viewports sized XL (extra-large) or wider.
    </p>
    <p
      className="text-xxl-left"
    >
      Left aligned text on viewports sized XXL (extra-extra-large) or wider.
    </p>
  </ComponentPreview>
</Fragment>
`;

exports[`Utilities: Text TextTransform renders 1`] = `
<Fragment>
  <h2
    id="text-transform"
  >
    Text transform
  </h2>
  <p>
    Transform text in components with text capitalization classes.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p
      className="text-lowercase"
    >
      Lowercased text.
    </p>
    <p
      className="text-uppercase"
    >
      Uppercased text.
    </p>
    <p
      className="text-capitalize"
    >
      CapiTaliZed text.
    </p>
  </ComponentPreview>
  <p>
    Note how 
    <CodeTags
      code=".text-capitalize"
      type="secondary"
    />
     only changes the first letter of each word, leaving the case of any other letters unaffected.
  </p>
</Fragment>
`;

exports[`Utilities: Text TextWrappingAndOverflow renders 1`] = `
<Fragment>
  <h2
    id="text-wrapping-and-overflow"
  >
    Text wrapping and overflow
  </h2>
  <p>
    Prevent text from wrapping with a 
    <CodeTags
      code=".text-nowrap"
      type="secondary"
    />
     class.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="text-nowrap sc-highlight"
      style={
        Object {
          "width": "8rem",
        }
      }
    >
      

      This text should overflow the parent.
      

    </div>
  </ComponentPreview>
  <p>
    For longer content, you can add a 
    <CodeTags
      code=".text-truncate"
      type="secondary"
    />
     class to truncate the text with an ellipsis. 
    <b>
      Requires
    </b>
     display: 
    <CodeTags
      code="inline-block"
      type="secondary"
    />
     
    <b>
      or
    </b>
     
    <CodeTags
      code="display: block"
      type="secondary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="row"
    >
      <div
        className="col-3 text-truncate"
      >
        Cras egestas dolor nec nunc scelerisque tincidunt.
      </div>
    </div>
    


    <span
      className="d-inline-block text-truncate"
      style={
        Object {
          "maxWidth": "150px",
        }
      }
    >
      

      Cras egestas dolor nec nunc scelerisque tincidunt.
      

    </span>
  </ComponentPreview>
</Fragment>
`;

exports[`Utilities: Text renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Documentation and examples for common text utilities to control alignment, wrapping, weight, and more.
  </p>
  <TextAlignment />
  <TextWrappingAndOverflow />
  <TextTransform />
  <FontWeightAndItalics />
</DocContainer>
`;
